contracts/NftInjectableTokenIdRolesUpdatableBaseURI.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract NftInjectableTokenIdRolesUpdatableBaseURI is ERC721URIStorage, AccessControl {
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    address private _owner;
+
+    bool public isFreezeTokenUris;
+    mapping (uint256 => bool) public freezeTokenUris;
+
+    string private baseURI;
```

Comment:
This is used when tokenURI function from ERC721URIStorage.sol is called yeah?

Reply:
check function _baseURI() - it overrides default function (which just returned "" by default)

Reply:
ah yes,  i see from tests

Reply:
Oh very cool nice